src/App.jsx

```diff
@@ -1,14 +1,27 @@
 import './App.css';
+import ChatLog from './components/ChatLog';
+import chatMessages from './data/messages.json';
+import { useState } from 'react';
 
 const App = () => {
+  const [messages, setMessages] = useState(chatMessages);
+
+  const toggleLike = (id) => {
+    const updatedMessages = messages.map((msg) =>
+      msg.id === id ? { ...msg, liked: !msg.liked } : msg
```

Comment:
We showed this approach in class, but technically, we're mixing a few responsibilities here. rather than this function needing to know how to change the liked status itself, we could move this update logic to a helper function. This would better mirror how we eventually update records when there's an API call involved.

In this project, our messages are very simple objects, but if we had more involved operations, it could be worthwhile to create an actual class with methods to work with them, or at least have a set of dedicated helper functions to centralize any such mutation logic.